File: jsi-rn-sys/src/ffi/console_host.cpp
#include "rust/cxx.h"
#include <jsi/jsi.h>
#include <react/runtime/TimerManager.h>
#include <PlatformTimerRegistryImpl.h>

void install_timer_globals(facebook::jsi::Runtime &runtime)
{
  auto platformTimers = std::make_unique<facebook::react::PlatformTimerRegistryImpl>();
  auto* platformTimersPtr = platformTimers.get();
  auto timerManager = std::make_shared<facebook::react::TimerManager>(std::move(platformTimers));
  platformTimersPtr->setTimerManager(timerManager);
  timerManager->attachGlobals(runtime);
}

File: jsi-rn-sys/src/ffi/console_host.rs
#[cxx::bridge]
pub(crate) mod console_host {

    unsafe extern "C++" {
        include!("console_host.h");

        #[namespace = "facebook::jsi"]
        type Runtime = jsi_sys::base::Runtime;

        pub fn install_timer_globals(runtime: Pin<&mut Runtime>) -> ();
    }
}

pub use console_host::*;
